src/sgame/sg_bot_util.cpp

```diff
@@ -904,6 +904,11 @@ bool BotChangeGoal( gentity_t *self, botTarget_t target )
 
 bool BotChangeGoalEntity( gentity_t *self, gentity_t const *goal )
 {
+	if ( !goal )
+	{
```

Comment:
`goal` is intentionally allowed to be null, meaning the bot has no goal. You can see that BotSetTarget handles this case.

Reply:
Then the crashes I had on that function might be due to other commits I made (which touch the goal stuff). I Will investigate more, I'll remove those commits from PR meanwhile.